### DISCUSSION TITLE
Battery Sensor to use for Energy Dashboard

### DISCUSSION THREAD

**Discussion by mr-birdie** [sfstar/hass-victron#199] [Q&A]
Good Day,

Thank you for this awesome integration. Loving it.

I am having a issue where I think not all my battery sensors are available for the energy dashboard.
These are the only ones showing, but they are reporting zero.

<img width="446" alt="image" src="https://user-images.githubusercontent.com/102876017/207609284-70affe85-89b6-4b95-b328-0163e9165ce5.png">

Is there something that I am missing? 

Also, can you maybe give the sensors that you use for your dashboard? All these sensors are quire confusing to me and I am not 100% sure which ones represent what.

Any help would be much appreciated

EDIT:

I have 282 entities available: 
<img width="301" alt="image" src="https://user-images.githubusercontent.com/102876017/207610157-d2fb3483-953a-432d-a876-891900aa921c.png">

And this is the firmware I am running (Updated to the latest possible)

![image](https://user-images.githubusercontent.com/102876017/207610308-f3508fb7-2df6-4dea-af2a-1b082796a2f1.png)

**Comment by sfstar** [comment]
Thank you for providing the additional information when opening this issue.
Currently I'm using the following entities to track my batteries usage:
<img width="380" alt="image" src="https://user-images.githubusercontent.com/6717280/207630260-c32e0448-90fa-4304-82d1-3facf558034c.png">

Whereby the acin2 to inverter doesn't see any current (my multiplus doesn't have an acin2).
Otherwise I couldn't add the feedin and the self consumption (Since HA doesn't allow for multiple in / out trackers in the same battery system).

Do you have an system with an can-bms battery? (Still waiting on shipping for my can-bms battery and currently using an voltage based highly inaccurate battery monitoring through the multiplusses battery monitoring).

Could you also provide me with a screenshot of your GX device available modbus devices? (under "menu" -> "settings" -> "services" -> "modbus tcp" -> "available services".
This will help in determining what entites should work best for battery tracking in your setup.

**Reply by MarcHagen** [reply]
anno 2025 i'm using `acin1toinverter` and `invertertoacin1` as one battery system in the Energy Dashboard.
(Im not using the AC out on the MP2)

**Comment by mr-birdie** [comment]
Thanks for the prompt response:

Below the modbus devices.
<img width="488" alt="image" src="https://user-images.githubusercontent.com/102876017/207654459-bfec8e64-db09-461d-9051-d4e6a3c82baf.png">

<img width="490" alt="image" src="https://user-images.githubusercontent.com/102876017/207654626-984ab7ce-beee-4cf3-b799-0ec5fc27550e.png">

System Overview:

<img width="505" alt="image" src="https://user-images.githubusercontent.com/102876017/207655382-73de8e5c-fe6e-456f-9520-6cd33c9e4543.png">

<img width="521" alt="image" src="https://user-images.githubusercontent.com/102876017/207655465-9871fd12-48a8-4f9a-9ece-e58f11a3ecf7.png">

Grid is currently of (due to loadshedding, perks of living in South Africa)

I assume AC-in means from Grid and AC-Out is the AC Loads of my house?

**Comment by sfstar** [comment]
Thank you for the additional information.
Yes the AC-in is `GRID/GENERATOR`
AC-OUT is `LOADS`

I just checked the victron modbus spec for any KWH based registers that you could use in your dashboard.
However, it seems that the battery unit doesn't expose KWH counters.
Currently (this might change if victron releases V3 of the gateway and the new modbus spec definition) your best option is to replicate my configuration with the vebus entities.

**Comment by CTRDevelopments** [comment]
Hi, I hand coded all of this (there was no repository so great work). You will need to integrate the energy stats for the energy page

Something like

`  #total energy to battery
  - platform: integration
    source: sensor.battery_input_energy
    name: total_battery_input_energy
    unit_prefix: k
    method: left
    #or right or trapezoidal for methods
    round: 3`

**Comment by mtucker502** [comment]
> Thank you for the additional information. Yes the AC-in is `GRID/GENERATOR` AC-OUT is `LOADS`
> 
> I just checked the victron modbus spec for any KWH based registers that you could use in your dashboard. However, it seems that the battery unit doesn't expose KWH counters. Currently (this might change if victron releases V3 of the gateway and the new modbus spec definition) your best option is to replicate my configuration with the vebus entities.

Does this include Lynx 1000A Shunt?
<img width="502" alt="image" src="https://user-images.githubusercontent.com/5040678/215866579-7a337cbf-b2e0-49ce-bf57-06619c8951cf.png">

<img width="508" alt="image" src="https://user-images.githubusercontent.com/5040678/215866699-8927d743-2920-4ccf-9c10-188fd7df4ba6.png">

It seems like the entities are present but have 0 values:
<img width="1052" alt="image" src="https://user-images.githubusercontent.com/5040678/215866962-2cf8acb4-a9b3-4ece-911e-b8f426979a39.png">

Or should I be using a different entity for tracking battery consumption?

**Comment by mtucker502** [comment]
> Thank you for providing the additional information when opening this issue. Currently I'm using the following entities to track my batteries usage: <img alt="image" width="380" src="https://user-images.githubusercontent.com/6717280/207630260-c32e0448-90fa-4304-82d1-3facf558034c.png">
> 
> Whereby the acin2 to inverter doesn't see any current (my multiplus doesn't have an acin2). Otherwise I couldn't add the feedin and the self consumption (Since HA doesn't allow for multiple in / out trackers in the same battery system).
> 
> Do you have an system with an can-bms battery? (Still waiting on shipping for my can-bms battery and currently using an voltage based highly inaccurate battery monitoring through the multiplusses battery monitoring).
> 
> Could you also provide me with a screenshot of your GX device available modbus devices? (under "menu" -> "settings" -> "services" -> "modbus tcp" -> "available services". This will help in determining what entites should work best for battery tracking in your setup.

Do these vebus sensors account for MPPT charge controllers and the power going straight into the batteries that doesn't pass through the inverters?

**Comment by mtucker502** [comment]
I have a BMS CAN battery (EG4 Communications Hub) and the battery charged/discharged values are not incrementing:

![image](https://user-images.githubusercontent.com/5040678/234416931-3aeb1412-b30d-4d56-aaf7-537daaa7fa77.png)

**Comment by RSickenberg** [comment]
Any updates?

**Comment by JohnMcLear** [comment]
I'm interested in this as well as it would be nice to get all my of data into the dashboard :) Thanks @sfstar et al for all your hard work ❤️
